internal/servers/controller/controller.go

```diff
@@ -119,32 +123,38 @@ func New(ctx context.Context, conf *Config) (*Controller, error) {
 		}
 	}
 
-	azureSvcClient, azureCleanup, err := external_host_plugins.CreateHostPlugin(
-		ctx,
-		"azure",
-		external_host_plugins.WithHostPluginsFilesystem("boundary-plugin-host-", host_plugin_assets.FileSystem()),
-		external_host_plugins.WithHostPluginExecutionDir(conf.RawConfig.Plugins.ExecutionDir),
-		external_host_plugins.WithLogger(hclog.NewNullLogger()))
-	if err != nil {
-		return nil, fmt.Errorf("error creating azure host plugin: %w", err)
-	}
-	conf.ShutdownFuncs = append(conf.ShutdownFuncs, azureCleanup)
-	if _, err := conf.RegisterHostPlugin(ctx, "azure", azureSvcClient, hostplugin.WithDescription("Built-in Azure host plugin")); err != nil {
-		return nil, fmt.Errorf("error registering azure host plugin: %w", err)
-	}
-
-	awsSvcClient, awsCleanup, err := external_host_plugins.CreateHostPlugin(
-		ctx,
-		"aws",
-		external_host_plugins.WithHostPluginsFilesystem("boundary-plugin-host-", host_plugin_assets.FileSystem()),
-		external_host_plugins.WithHostPluginExecutionDir(conf.RawConfig.Plugins.ExecutionDir),
-		external_host_plugins.WithLogger(hclog.NewNullLogger()))
-	if err != nil {
-		return nil, fmt.Errorf("error creating aws host plugin")
-	}
-	conf.ShutdownFuncs = append(conf.ShutdownFuncs, awsCleanup)
-	if _, err := conf.RegisterHostPlugin(ctx, "aws", awsSvcClient, hostplugin.WithDescription("Built-in AWS host plugin")); err != nil {
-		return nil, fmt.Errorf("error registering aws host plugin: %w", err)
+	for _, enabledPlugin := range c.enabledPlugins {
+		switch enabledPlugin {
+		case base.EnabledPluginHostAzure:
```

Comment:
These 2 case statements seem very similar.  Consider containing the plugin information (like the name and... is that it?) with the EnabledPluginHost type so there is a centralized place for that info and this code can avoid being duplicated.

Reply:
Good idea, fixed!